crates/locutus-node/src/operations/put.rs

```diff
@@ -426,6 +436,65 @@ where
                 return Err(OpError::StatePushed);
             }
         }
+        PutMsg::BroadcastTo {
+            id,
+            key,
+            new_value,
+            sender,
+            sender_subscribers,
+        } => {
+            let target = op_storage.ring.own_location();
+
+            log::debug!("Attempting contract value update");
+            let new_value = put_contract(op_storage, key, new_value).await?;
+            log::debug!("Contract successfully updated");
+
+            let broadcast_to = op_storage
+                .ring
+                .subscribers_of(&key)
+                .map(|i| {
+                    // Avoid already broadcast nodes and sender from broadcasting
+                    let mut subscribers: Vec<PeerKeyLocation> = i.value().to_vec();
+                    let mut avoid_list: HashSet<PeerKey> =
+                        sender_subscribers.into_iter().map(|pl| pl.peer).collect();
+                    avoid_list.insert(sender.peer);
+                    subscribers.retain(|s| !avoid_list.contains(&s.peer));
+                    subscribers
+                })
+                .unwrap_or_default();
+            log::debug!(
+                "Successfully updated a value for contract {} @ {:?}",
+                key,
+                target.location
+            );
+
+            let internal_cb = state
+                .sm
+                .consume_to_output(PutMsg::Broadcasting {
+                    id,
+                    broadcast_to,
+                    broadcasted_to: 0,
+                    key,
+                    new_value,
+                })?
+                .ok_or(OpError::InvalidStateTransition(id))?;
+
+            if let PutMsg::SuccessfulUpdate { .. } = internal_cb {
+                log::debug!(
+                    "Empty broadcast list while updating value for contract {}",
+                    key
+                );
+                // means the whole tx finished so can return early
+                return_msg = Some(internal_cb.into());
+                new_state = None;
+            } else {
+                log::debug!("Notify nodes to to broadcast");
```

Comment:
typo: double "to"
Also more correct would be "callback to start broadcasting to other nodes"